Relative paths in the `boost_inc_dir` are adjusted to be relative to `build` rather than the source root on Windows

<!-- Please fill in the following information to report a problem with Cantera. If you have a question about using Cantera, please post it on our Google Users' Group (https://groups.google.com/forum/#!forum/cantera-users). Feature enhancements should be discussed in the dedicated Cantera enhancements repository (https://github.com/Cantera/enhancements/new/choose) -->

**Problem description**

If Boost is specified with a relative directory in the `boost_inc_dir` option, SCons adjusts the include path to be relative to the `build` and `src` directories when writing command lines for compiling. For example:

```
cl /Fobuild\src\base\AnyMap.obj /c src\base\AnyMap.cpp /TP /EHsc /std:c++17 /utf-8 /MD /nologo /D_SCL_SECURE_NO_WARNINGS /D_CRT_SECURE_NO_WARNINGS /O2 /Zi /Fdbuild\src\base\AnyMap.obj.pdb /W3 /DNDEBUG /Iinclude\cantera\ext /Iinclude /Ibuild\src /Ibuild\src\boost.1.85.0\lib\native\include /Isrc\boost.1.85.0\lib\native\include
AnyMap.cpp
C:\Users\Quickemu\Documents\cantera\src\base\application.h(12): fatal error C1083: Cannot open include file: 'boost/algorithm/string/join.hpp': No such file or directory
scons: *** [build\src\base\AnyMap.obj] Error 2
```

Note the `/Ibuild\src\boost.1.85.0\lib\native\include`. In `cantera.conf`, I set `boost_inc_dir = 'boost.1.85.0/lib/native/include'` and SCons added the `build\src` prefix. This seems to only happen on Windows, Linux seems to use a relative path to the `cantera.conf` file, as expected.

Changing
```
 env["boost_inc_dir"] = Path(env["boost_inc_dir"]).as_posix()
```
to
```
 env["boost_inc_dir"] = Path(env["boost_inc_dir"]).resolve().as_posix()
```
fixes the issue and doesn't seem to break anything on Linux. This change should also be made to other include and library config options for consistency.

--- a/SConstruct
+++ b/SConstruct
@@ -1036,10 +1036,37 @@ elif env['env_vars']:
             logger.warning(f"Failed to propagate environment variable {name!r}\n"
                            "Edit cantera.conf or the build command line to fix this.")
 
-env["extra_inc_dirs"] = [Path(d).as_posix() for d in env["extra_inc_dirs"].split(os.pathsep) if d]
-env["extra_lib_dirs"] = [Path(d).as_posix() for d in env["extra_lib_dirs"].split(os.pathsep) if d]
+
+def make_relative_path_absolute(path_to_check: str | Path) -> str:
+    """If a path is absolute, return it in POSIX format.
+    If a path is relative, assume it's relative to the source root, convert it to an
+    absolute path, and return the converted path in POSIX format.
+    """
+    pth = Path(path_to_check)
+    if not pth.is_absolute():
+        pth = Path(Dir("#" + path_to_check).abspath)
+
+    return pth.as_posix()
+
+
+inc_dirs = []
+for inc_dir in env["extra_inc_dirs"].split(os.pathsep):
+    if not inc_dir:
+        continue
+    inc_dirs.append(make_relative_path_absolute(inc_dir))
+
+env["extra_inc_dirs"] = inc_dirs
+
+lib_dirs = []
+for lib_dir in env["extra_lib_dirs"].split(os.pathsep):
+    if not lib_dir:
+        continue
+    lib_dirs.append(make_relative_path_absolute(lib_dir))
+
+env["extra_lib_dirs"] = lib_dirs
 
 # Add conda library/include paths (if applicable) to extra
+# Assume "CONDA_PREFIX" is always absolute if it exists
 conda_prefix = os.environ.get("CONDA_PREFIX")
 if conda_prefix is not None:
     conda_prefix = Path(conda_prefix)
@@ -1070,22 +1097,22 @@ else:
     env['FORTRAN_LINK'] = '$FORTRAN'
 
 if env['boost_inc_dir']:
-    env["boost_inc_dir"] = Path(env["boost_inc_dir"]).as_posix()
+    env["boost_inc_dir"] = make_relative_path_absolute(env["boost_inc_dir"])
     add_system_include(env, env['boost_inc_dir'])
 
 if env['blas_lapack_dir']:
-    env["blas_lapack_dir"] = Path(env["blas_lapack_dir"]).as_posix()
+    env["blas_lapack_dir"] = make_relative_path_absolute(env["blas_lapack_dir"])
     env.Append(LIBPATH=[env['blas_lapack_dir']])
     if env['use_rpath_linkage']:
         env.Append(RPATH=env['blas_lapack_dir'])
 
 if env['system_sundials'] in ('y','default'):
     if env['sundials_include']:
-        env["sundials_include"] = Path(env["sundials_include"]).as_posix()
+        env["sundials_include"] = make_relative_path_absolute(env["sundials_include"])
         add_system_include(env, env['sundials_include'])
         env['system_sundials'] = 'y'
     if env['sundials_libdir']:
-        env["sundials_libdir"] = Path(env["sundials_libdir"]).as_posix()
+        env["sundials_libdir"] = make_relative_path_absolute(env["sundials_libdir"])
         env.Append(LIBPATH=[env['sundials_libdir']])
         env['system_sundials'] = 'y'
         if env['use_rpath_linkage']:

--- a/SConstruct
+++ b/SConstruct
@@ -1037,18 +1037,6 @@ elif env['env_vars']:
                            "Edit cantera.conf or the build command line to fix this.")
 
 
-def make_relative_path_absolute(path_to_check: str | Path) -> str:
-    """If a path is absolute, return it in POSIX format.
-    If a path is relative, assume it's relative to the source root, convert it to an
-    absolute path, and return the converted path in POSIX format.
-    """
-    pth = Path(path_to_check)
-    if not pth.is_absolute():
-        pth = Path(Dir("#" + path_to_check).abspath)
-
-    return pth.as_posix()
-
-
 inc_dirs = []
 for inc_dir in env["extra_inc_dirs"].split(os.pathsep):
     if not inc_dir:

--- a/site_scons/buildutils.py
+++ b/site_scons/buildutils.py
@@ -27,7 +27,7 @@ __all__ = ("Option", "PathOption", "BoolOption", "EnumOption", "Configuration",
            "add_RegressionTest", "get_command_output", "listify", "which",
            "ConfigBuilder", "multi_glob", "get_spawn", "quoted", "add_system_include",
            "get_pip_install_location", "compiler_flag_list", "setup_python_env",
-           "checkout_submodule", "check_for_python")
+           "checkout_submodule", "check_for_python", "make_relative_path_absolute")
 
 if TYPE_CHECKING:
     from typing import Iterable, TypeVar, Union, List, Dict, Tuple, Optional, \
@@ -1631,3 +1631,15 @@ def check_for_python(env: SCEnvironment, command_line_targets: List[str]) -> Dic
             logger.info(f"Using pytest version {pytest_version}")
 
     return {"python_package": "y", "require_numpy_1_7_API": require_numpy_1_7_API}
+
+
+def make_relative_path_absolute(path_to_check: Union[str, Path]) -> str:
+    """If a path is absolute, return it in POSIX format.
+    If a path is relative, assume it's relative to the source root, convert it to an
+    absolute path, and return the converted path in POSIX format.
+    """
+    pth = Path(path_to_check)
+    if not pth.is_absolute():
+        pth = Path(Dir("#" + path_to_check).abspath)
+
+    return pth.as_posix()

